tests/conftest.py

```diff
@@ -205,3 +207,29 @@ def single_version():
 @pytest.fixture(scope="function")
 def no_numpy_version():
     return {"python": ["2.7.*", "3.5.*"]}
+
+
+@pytest.fixture(scope="function")
+def variants_conda_build_sysroot(monkeypatch):
+    if not on_mac:
+        return {}
+
+    monkeypatch.setenv(
+        "CONDA_BUILD_SYSROOT",
+        subprocess.run(
+            ["xcrun", "--sdk", "macosx", "--show-sdk-path"],
+            check=True,
+            capture_output=True,
+            text=True,
+        ).stdout.strip(),
+    )
+    monkeypatch.setenv(
+        "MACOSX_DEPLOYMENT_TARGET",
+        subprocess.run(
+            ["xcrun", "--sdk", "macosx", "--show-sdk-version"],
+            check=True,
+            capture_output=True,
+            text=True,
+        ).stdout.strip(),
+    )
+    return {"MACOSX_DEPLOYMENT_TARGET": ["10.9"]}
```

Comment:
Maybe this works for retaining the two cases?
(Untested; I never used `fixtures`+`params` before...)
```suggestion
@pytest.fixture(scope="function", params=[{}, {"MACOSX_DEPLOYMENT_TARGET": ["10.9"]})
def variants_conda_build_sysroot(monkeypatch, request):
    if not on_mac:
        return {}

    monkeypatch.setenv(
        "CONDA_BUILD_SYSROOT",
        subprocess.run(
            ["xcrun", "--sdk", "macosx", "--show-sdk-path"],
            check=True,
            capture_output=True,
            text=True,
        ).stdout.strip(),
    )
    monkeypatch.setenv(
        "MACOSX_DEPLOYMENT_TARGET",
        subprocess.run(
            ["xcrun", "--sdk", "macosx", "--show-sdk-version"],
            check=True,
            capture_output=True,
            text=True,
        ).stdout.strip(),
    )
    return request.param
```

Reply:
Ah! This makes more sense! This is an improvement over the previous test cases since we're properly testing not overriding envvars and then overriding.

Since this is specific to macOS I think we need an additional tweak:

```suggestion
@pytest.fixture(
    scope="function",
    params=[{}, {"MACOSX_DEPLOYMENT_TARGET": ["10.9"]}] if on_mac else [{}],
)
def variants_conda_build_sysroot(monkeypatch, request):
    if not on_mac:
        return {}

    monkeypatch.setenv(
        "CONDA_BUILD_SYSROOT",
        subprocess.run(
            ["xcrun", "--sdk", "macosx", "--show-sdk-path"],
            check=True,
            capture_output=True,
            text=True,
        ).stdout.strip(),
    )
    monkeypatch.setenv(
        "MACOSX_DEPLOYMENT_TARGET",
        subprocess.run(
            ["xcrun", "--sdk", "macosx", "--show-sdk-version"],
            check=True,
            capture_output=True,
            text=True,
        ).stdout.strip(),
    )
    return request.param
```